turbopack/crates/turbopack-ecmascript/tests/tree-shaker/analyzer/dce/output.md

```diff
@@ -0,0 +1,218 @@
+# Items
+
+Count: 4
+
+## Item 1: Stmt 0, `ImportOfModule`
+
+```js
+import { baz } from './module';
+
+```
+
+- Hoisted
+- Side effects
+
+## Item 2: Stmt 0, `ImportBinding(0)`
+
+```js
+import { baz } from './module';
+
+```
+
+- Hoisted
+- Declares: `baz`
+
+## Item 3: Stmt 1, `Normal`
+
+```js
+if (1 + 1 == 3) {
+    baz();
+}
+
+```
+
+- Side effects
+- Reads: `baz`
+
+# Phase 1
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+```
+# Phase 2
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+    Item3 --> Item2;
+    Item3 --> Item1;
+```
+# Phase 3
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+    Item3 --> Item2;
+    Item3 --> Item1;
+```
+# Phase 4
+```mermaid
+graph TD
+    Item1;
+    Item2;
+    Item3;
+    Item4;
+    Item4["ModuleEvaluation"];
+    Item3 --> Item2;
+    Item3 --> Item1;
+    Item4 --> Item1;
+    Item4 --> Item3;
+```
+# Final
+```mermaid
+graph TD
+    N0["Items: [ItemId(0, ImportOfModule)]"];
+    N1["Items: [ItemId(0, ImportBinding(0))]"];
+    N2["Items: [ItemId(1, Normal)]"];
+    N3["Items: [ItemId(ModuleEvaluation)]"];
+    N2 --> N1;
+    N2 --> N0;
+    N3 --> N0;
+    N3 --> N2;
+```
```

Comment:
Follow-up change:

The whole graph has only a single entrypoint `ModuleEvaluation`, so it should result in a single fragment/part. The graph reduction isn't working as efficient as it could.

Reply:
https://linear.app/vercel/issue/PACK-3286/